### PR TITLE
Trying other Python versions

### PR DESCRIPTION
**Pull request by wpietri**
Making the PR mainly to trigger workflows.

### CODE REVIEW

**Comment by wpietri** [comment]
Ok, now this actually works. Tests are run in the correct python versions.

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -12,20 +12,24 @@ jobs:
...
-        python-version: "3.10"
+        python-version: ${{ matrix.python-version }}
         
     - name: cache poetry install
       uses: actions/cache@v3
       id: cache-poetry
       with:
         path: ~/.local
-        key: poetry-1.8.2-0
+        key: poetry-1.8.2-${{ steps.setup-python.outputs.python-version }}-0
```
Comment (yifanmai) [inline .github/workflows/python-app.yml:32]: I'd suggest using the [built-in Poetry cache support in `actions/setup-python`](https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies) instead, which should be less brittle. That should allow you to drop the `actions/cache` steps entirely.

Comment (wpietri) [reply .github/workflows/python-app.yml:32]: Ooh, neat! Thanks for the tip.

**Comment by wpietri** [comment]
Ok @dhosterman, I think this is at last ready to review. Thanks @yifanmai for the simplification tip.

**Review by dhosterman — APPROVED** [review]
👍

**Pull request merged**